float64 left_x
float64 right_x
float64 left_theta
float64 right_theta